analysis/quantifier-comparisons/09-emptydrop-comparisons.Rmd

```diff
@@ -0,0 +1,461 @@
+colnames(coldata_cellranger) <- stringr::str_replace(colnames(coldata_cellranger), "^subsets_mito_", "mito_")
+
+# create one coldata df with both alevin fry and cellranger, all filtering methods
+coldata_df <- dplyr::bind_rows(fry_coldata_df, coldata_cellranger)
+```
+
+
+```{r fig.height=10, fig.width=10}
+ggplot(coldata_df, aes(x = tool, y = sum, fill = filtering_method)) + 
+  geom_boxplot() + 
+  facet_wrap(~quant_id, nrow = 2) + 
+  coord_cartesian(ylim = c(0,25000)) + 
+  ylab("UMI/cell")
+```
+
+```{r fig.height=10, fig.width=10}
+ggplot(coldata_df, aes(x = tool, y = detected, fill = filtering_method)) + 
+  geom_boxplot() + 
+  facet_wrap(~quant_id, nrow = 2) + 
+  coord_cartesian(ylim = c(0,7500)) + 
+  ylab("Genes detected/cell")
+```
+Here you can see that the median UMI/cell and genes detected/cell look lower for Alevin-fry with `emptyDrops` than Cell Ranger, which is consistent with the previous finding that they are reporting more cells than Cell Ranger. 
+However, if we use that threshold of `lower=200` or `lower=500` for `emptyDrops`, we see a much more similar distribution and median of UMI/cell and genes detected/cell. 
+
+The last check that we'll do is see how many of the cell barcodes are overlapping between the `emptyDrops` `lower=200` filtered sces and the Cell Ranger sces. 
+
+```{r}
+percent_filter_overlap <- function(filter_sce, cellranger_sce){
+  overlap <- length(intersect(colnames(filter_sce), 
+                              colnames(cellranger_sce)))
+  total_filter <- length(colnames(filter_sce))
+  total_cellranger <- length(colnames(cellranger_sce))
+  percent_filter <- round(overlap/total_filter * 100,2)
+  percent_cellranger <- round(overlap/total_cellranger * 100,2)
+  glue::glue("Percent of filtered cell barcodes also in Cell Ranger {percent_filter} \nPercent of Cell Ranger cell barcodes also in filtered {percent_cellranger}")
+}
+
+purrr::map2(emptydrops_filter_sces_500, 
+            cellranger_sces[names(emptydrops_filter_sces_500)], 
+            percent_filter_overlap)
+```
+
+```{r}
+purrr::map2(emptydrops_filter_sces_200, 
+            cellranger_sces[names(emptydrops_filter_sces_200)], 
+            percent_filter_overlap)
+```
+
+It appears that the majority if the cell barcodes in Cell Ranger are also found in our filtered sces (which is a good sign), and over 75% of the cell barcodes that we see in the filtered sce are also found in Cell Ranger for both `lower=200` and `lower=500`. 
+
+```{r}
+# Do the same test but with the emptyDrops lower = 100
+purrr::map2(emptydrops_filter_sces, 
+            cellranger_sces[names(emptydrops_filter_sces)], 
+            percent_filter_overlap)
+```
+In contrast, only 50% of the barcodes that are output from Alevin-fry filtered by `emptyDrops` with `lower=100` are found in Cell Ranger. 
+It appears that lowering the threshold of minimum UMI count helps retain a more similar group of cells to Cell Ranger. 
```

Comment:
That was not my interpretation; To me 200 looked much better than 100, if I am interpreting the text correctly. There is one case (118) where cellranger is retaining many more cells, but as previously discussed, I don't think that data is useful anyway. Otherwise, there is usually better overlap at the 200 level.

Reply:
This was a typo, I agree that both 200 and 500 looked much better than 100. I fixed the text accordingly.